### DISCUSSION TITLE
Google OAuth Does Not Return Separate First Name and Last Name Fields

### DISCUSSION THREAD

**Discussion by theodorosmd** [supabase/supabase#28415] [Feature Requests]
Description:

I'm using Supabase for authentication in my project and have integrated Google OAuth for user sign-in. However, I have encountered an issue where the user's first name and last name are not returned as separate fields in the authentication response. Instead, the full name is returned as a single string.

Steps to Reproduce:

Set up a Supabase project.
Integrate Google OAuth following the official documentation.
Authenticate using a Google account.

Expected Behavior:

The authentication response should include separate fields for the user's first name and last name, allowing for easier manipulation and storage of user information.

Actual Behavior:

The authentication response includes a single name field containing the user's full name. This requires additional parsing to separate the first name and last name.

Impact:

This issue complicates user management and data handling, especially in applications that require distinct first and last names for personalization, form filling, and other purposes.

Suggested Solution:

It would be beneficial if Supabase could modify the Google OAuth integration to return the user's first name and last name as separate fields. This change would align with the data structure provided by many other OAuth providers and simplify user data handling.

Thank you for your attention to this matter.

**Comment by CRREO** [comment]
> Description:
> 
> I'm using Supabase for authentication in my project and have integrated Google OAuth for user sign-in. However, I have encountered an issue where the user's first name and last name are not returned as separate fields in the authentication response. Instead, the full name is returned as a single string.
> 
> Steps to Reproduce:
> 
> Set up a Supabase project. Integrate Google OAuth following the official documentation. Authenticate using a Google account.
> 
> Expected Behavior:
> 
> The authentication response should include separate fields for the user's first name and last name, allowing for easier manipulation and storage of user information.
> 
> Actual Behavior:
> 
> The authentication response includes a single name field containing the user's full name. This requires additional parsing to separate the first name and last name.
> 
> Impact:
> 
> This issue complicates user management and data handling, especially in applications that require distinct first and last names for personalization, form filling, and other purposes.
> 
> Suggested Solution:
> 
> It would be beneficial if Supabase could modify the Google OAuth integration to return the user's first name and last name as separate fields. This change would align with the data structure provided by many other OAuth providers and simplify user data handling.
> 
> Thank you for your attention to this matter.

hi @theodorosmd - any chance you came to a solution for this? Thank you

**Reply by hecker** [reply]
Any updates here?

**Reply by XStarlink** [reply]
Hello @GaryAustin1, I apologize for pinging you here — I wasn’t sure who else on the Supabase team might see this discussion.
I'm experiencing the same issue: Google OAuth returns only a single full_name field, without separate first and last names. This makes user handling difficult for emailing, invoicing, and many other cases.

Has there been any update or workaround for this?

Thank you very much for your help!

**Reply by GaryAustin1** [reply]
I believe Supabase does not process what is returned from the OAuth provider. Just adds it to metadata.  I’ve not looked at their source in awhile on this.

**Reply by XStarlink** [reply]
> I believe Supabase does not process what is returned from the OAuth provider. Just adds it to metadata. I’ve not looked at their source in awhile on this.

Thank you for the clarification! I’ll continue my research, but so far I haven’t found any more information on this. I also looked at the code [here](https://github.com/supabase/auth/blob/dbaccd4dbb1ca54c603d84ab201b07df822f7771/internal/api/provider/google.go#L13), and it seems that the given_name and family_name fields returned by Google are not included there.

**Reply by GaryAustin1** [reply]
Looks like it uses only the Name value from Google.

**Reply by XStarlink** [reply]
Yes, exactly — it would be really useful if Supabase could standardize this across all providers. Whenever the provider returns given_name and family_name, having them mapped consistently to first_name and last_name in Supabase would make it much easier to rely on those fields, especially for SaaS apps (e.g. emailing, invoicing, personalization).

Do you know who on the Supabase team would be the right person to talk to about proposing this as a feature request?

**Reply by hecker** [reply]
Cannot agree more with @XStarlink. We have a function in our database to make sure to always extract to `first_name` and `last_name` if not available from OAuth providers.

**Reply by GaryAustin1** [reply]
@XStarlink It would be an supabase/auth feature.  They only have an issues section there.  
I don't know how often Supabase reads here (I'm not an employee).  I only jumped in as was pinged.
